scaladoc/src/dotty/tools/scaladoc/SocialLinks.scala

```diff
@@ -5,11 +5,15 @@ enum SocialLinks(val url: String, val className: String):
   case Twitter(tUrl: String) extends SocialLinks(tUrl, "twitter")
   case Gitter(gUrl: String) extends SocialLinks(gUrl, "gitter")
   case Discord(dUrl: String) extends SocialLinks(dUrl, "discord")
+  case Custom(cUrl: String, lightIcon: String, darkIcon: String) extends SocialLinks(cUrl, "custom")
 
 object SocialLinks:
+  val LowercaseNamePattern = "^[a-z]+$".r
```

Comment:
I think since we're introducing some more parsing here do you think you could add a small unit test to just ensure the logic for `parse` works as expected?